src/test/java/com/mapzen/valhalla/HttpHandlerTest.java

```diff
@@ -0,0 +1,77 @@
+package com.mapzen.valhalla;
+
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.mockito.internal.util.reflection.Whitebox;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static com.mapzen.TestUtils.getFixture;
+import static org.fest.assertions.api.Assertions.assertThat;
+import retrofit.Endpoint;
+import retrofit.RequestInterceptor;
+import retrofit.RestAdapter;
+
+public class HttpHandlerTest {
+
+  String endpoint = "https://valhalla.mapzen.com/";
+  HttpHandler httpHandler;
+
+  @Before public void setup() throws IOException {
+    httpHandler = new HttpHandler(endpoint, RestAdapter.LogLevel.NONE);
+  }
+
+  @Test public void shouldHaveEndpoint() {
+    final String endpoint =
+        (String) Whitebox.getInternalState(httpHandler, "endpoint");
+    assertThat(endpoint).startsWith("https://valhalla.mapzen.com/");
+  }
+
+  @Test public void shouldHaveLogLevel() {
+    final RestAdapter.LogLevel logLevel =
+        (RestAdapter.LogLevel) Whitebox.getInternalState(httpHandler, "logLevel");
+    assertThat(logLevel).isEqualTo(RestAdapter.LogLevel.NONE);
+  }
+
+  @Test public void shouldAddHeaders() throws IOException {
+    final MockWebServer server = new MockWebServer();
+    server.start();
+    server.enqueue(new MockResponse().setBody(getFixture("brooklyn")));
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
+    executorService.execute(new Runnable() {
+      @Override
+      public void run() {
+        String endpoint = server.getUrl("").toString();
+        TestHttpHandler httpHandler = new TestHttpHandler(endpoint, RestAdapter.LogLevel.NONE);
+        Router router = new ValhallaRouter()
+            .setHttpHandler(httpHandler)
+            .setLocation(new double[] { 40.659241, -73.983776 })
+            .setLocation(new double[] { 40.671773, -73.981115 });
+        RouteCallback callback = Mockito.mock(RouteCallback.class);
+        router.setCallback(callback);
+        router.fetch();
+        assertThat(httpHandler.headersAdded).isTrue();
```

Comment:
Unfortunately there is a problem with placing assertions in a `Runnable` in that even if the assertion fails the test will still pass. I think there may be some legacy tests that used this method before we discovered the issue.

A better approach for testing the router is to call `ValhallaRouter#run()` directly instead of `fetch()` which will execute the code in the main thread. For example see this test:

https://github.com/mapzen/on-the-road/blob/master/src/test/java/com/mapzen/valhalla/RouterTest.java#L311